error.go

```diff
@@ -25,41 +25,37 @@ var _ fmt.Formatter = (*Error)(nil)
 // If an error already contained another value for the same property, it is overwritten.
 // It is a caller's responsibility to accumulate and update a property, if needed.
 // Dynamic properties is a brittle mechanism and should therefore be used with care and in a simple and robust manner.
+// Currently, properties are implemented as a linked list, therefore it is not safe to have many dozens of them. But couple of dozen is just ok.
```

Comment:
It is safe actually. Maybe just not as performant as it could be with map.

Reply:
I never consider `O(n^2)` as a safe :-)

Reply:
The documentation should be clear though. `WithProperty` will behave correctly even with hundred million dozen properties. But the performance wouldn't be pleasant. You should mention that `WithProperty` costs O(N) instead of declaring it as unsafe.